File: PR Kelas/18 mar/Mobil.cpp
#include "Mobil.hpp"

Mobil::Mobil() : Kendaraan() {
    this->kategori="mobil";
    this->supir = "XXXX";
}

Mobil::Mobil(int nomor, int tahun_keluaran, string merk, string kategori="mobil", string supir) : Kendaraan(nomor, tahun_keluaran, merk, kategori) {
    this->supir=supir;
}

Mobil::Mobil(const Mobil &m) : Kendaraan(m) {
    this->supir=m.supir;
}

Mobil& Mobil::operator=(const Mobil& m) {
    if (this != &m) {
        this->nomor=m.nomor;
        this->tahun_keluaran=m.tahun_keluaran;
        this->merk=m.merk;
        this->kategori=m.kategori;
        this->supir=m.supir;
    }
    return *this;
} 

void Mobil::printInfo() {
    Kendaraan::printInfo();
    cout << "Supir: " << this->supir << endl;
}

void Mobil::biayaSewa(int lamaSewa){
    int hargaSewa=500000*lamaSewa;
    cout << "Biaya sewa kendaraan ini adalah: " << hargaSewa << endl;
}

File: PR Kelas/13 mei/Mobil.java
class Mobil extends Kendaraan {
    private String supir;
    public Mobil() {
        super();
        this.kategori="mobil";
        this.supir="XXX";
    }
    public Mobil(int n, int t, String m, String k, String s) {
        super(n, t, m, k);
        this.supir=s;
    }
    public Mobil(Mobil k) {
        this.nomor=k.nomor;
        this.tahun_keluaran=k.tahun_keluaran;
        this.merk=k.merk;
        this.kategori=k.kategori;
        this.supir=k.supir;
    }
    // Tidak ada operator overloading
    @Override
    public void printInfo() {
        super.printInfo();
        System.out.println("Supir: " + this.supir);
    }
    @Override
    public int biayaSewa(int l) {
        int hargaSewa=500000*l;
        System.out.println("Biaya sewa kendaraan ini adalah: " + hargaSewa);
        return hargaSewa;
    }
}